pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceTest.java

```diff
@@ -2002,5 +2014,39 @@ public void testTlsWithAuthParams() throws Exception {
         }
     }
 
+    @Test
+    public void testPendingPublishBufferUsageMetrics() throws Exception {
+        PulsarTestContext pulsarTestContext = this.pulsarTestContext;
+        InMemoryMetricReader reader = pulsarTestContext.getOpenTelemetryMetricReader();
+
+        Collection<MetricData> metricData = reader.collectAllMetrics();
+        OpenTelemetryAssertions.assertThat(metricData).anySatisfy(metric ->
+                OpenTelemetryAssertions.assertThat(metric).hasName("pulsar.broker.publish.buffer.usage")
+                        .hasLongGaugeSatisfying(gauge ->
+                                gauge.hasPointsSatisfying(point -> point.hasValue(0))));
+        OpenTelemetryAssertions.assertThat(metricData).anySatisfy(metric ->
+                OpenTelemetryAssertions.assertThat(metric).hasName("pulsar.broker.publish.buffer.max.usage")
+                        .hasLongGaugeSatisfying(gauge ->
+                                gauge.hasPointsSatisfying(point -> point.hasValue(0))));
+
+        String topic = "persistent://prop/ns-abc/testPendingPublishBufferUsageMetrics";
+        @Cleanup
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING).topic(topic).enableBatching(false).create();
+        for (int i = 0; i < 10; i++) {
+            producer.send("test");
+        }
+
+        metricData = reader.collectAllMetrics();
+        OpenTelemetryAssertions.assertThat(metricData).anySatisfy(metric ->
+                OpenTelemetryAssertions.assertThat(metric).hasName("pulsar.broker.publish.buffer.usage")
+                        .hasLongGaugeSatisfying(gauge ->
+                                gauge.hasPointsSatisfying(point -> point.hasValue(0))));
```

Comment:
The initial value and result are both 0, it cannot be determined whether it has taken effect?